libs/cluster/Session/ClusterCommands.cs

```diff
@@ -1055,38 +1144,46 @@ public bool ProcessSlotManageCommands(ReadOnlySpan<byte> bufSpan, ReadOnlySpan<b
                     }
 
                     //Parse slot ranges
-                    if (!ParseSlots(_count, ref ptr, out var slots, out var resp, range: true))
+                    if (!TryParseSlots(_count, ref ptr, out var slots, out var errorMessage, range: true))
                         return false;
+
                     readHead = (int)(ptr - recvBufferPtr);
 
                     //Execute
-                    if (resp.SequenceEqual(CmdStrings.RESP_OK))
+                    if (errorMessage == default)
                     {
                         switch (slotState)
                         {
                             case SlotState.STABLE:
-                                _ = clusterProvider.clusterManager.ResetSlotsState(slots, out resp);
+                                _ = clusterProvider.clusterManager.TryResetSlotsState(slots, out errorMessage);
                                 break;
                             case SlotState.IMPORTING:
-                                _ = clusterProvider.clusterManager.PrepareSlotsForImport(slots, nodeid, out resp);
+                                _ = clusterProvider.clusterManager.TryPrepareSlotsForImport(slots, nodeid, out errorMessage);
                                 break;
                             case SlotState.MIGRATING:
-                                _ = clusterProvider.clusterManager.PrepareSlotsForMigration(slots, nodeid, out resp);
+                                _ = clusterProvider.clusterManager.TryPrepareSlotsForMigration(slots, nodeid, out errorMessage);
                                 break;
                             case SlotState.NODE:
-                                _ = clusterProvider.clusterManager.PrepareSlotsForOwnershipChange(slots, nodeid, out resp);
+                                _ = clusterProvider.clusterManager.TryPrepareSlotsForOwnershipChange(slots, nodeid, out errorMessage);
                                 break;
                             default:
-                                resp = Encoding.ASCII.GetBytes($"-ERR Slot state {subcommand} not supported.\r\n");
-                                while (!RespWriteUtils.WriteDirect(resp, ref dcurr, dend))
-                                    SendAndReset();
+                                errorMessage = Encoding.ASCII.GetBytes($"ERR Slot state {subcommand} not supported.");
                                 break;
                         }
                     }
 
-                    if (resp.SequenceEqual(CmdStrings.RESP_OK)) UnsafeWaitForConfigTransition();
-                    while (!RespWriteUtils.WriteDirect(resp, ref dcurr, dend))
-                        SendAndReset();
+                    if (errorMessage == default)
```

Comment:
nit; use returned boolean from above to check for errors.

Reply:
[Addressed](https://github.com/microsoft/garnet/pull/252#discussion_r1556396125)